tests/test_misc.py

```diff
@@ -85,27 +85,30 @@ def test_explicit_missing_cache_entries(mocker):
     """Test that explicit() raises and notifies if some of the specs were not found in the cache."""
     from conda.core.package_cache_data import PackageCacheData
 
-    if len(PackageCacheData.get_all_extracted_entries()) == 0:
-        # Package cache e.g. ./devenv/Darwin/x86_64/envs/devenv-3.9-c/pkgs/ can
-        # be empty in certain cases (Noted in OSX with Python 3.9, when
-        # Miniconda installs Python 3.10). Install a small package.
-        warnings.warn("test_explicit_missing_cache_entries: No packages in cache.")
-        run_command(Commands.INSTALL, context.target_prefix, "heapdict")
-
-    # Patching ProgressiveFetchExtract prevents trying to download a package from the url.
-    # Note that we cannot monkeypatch context.dry_run, because explicit() would exit early with that.
-    mocker.patch("conda.misc.ProgressiveFetchExtract")
-
-    with pytest.raises(
-        AssertionError, match="Missing package cache records for: pkgs/linux-64::foo==1.0.0=py_0"
-    ):
-        explicit(
-            [
-                "http://test/pkgs/linux-64/foo-1.0.0-py_0.tar.bz2",  # does not exist
-                PackageCacheData.get_all_extracted_entries()[0].url,  # exists
-            ],
-            "",
-        )
+    with make_temp_env() as prefix:  # ensure writable env
```

Comment:
Another order dependency. After enough tests have run, `context.target_prefix` has packages; before enough tests have run, the read-only /opt/conda in the tests container has no packages. ?? so how does it get packages later during the normal test runs, since it is read-only? (we do rm -rf the package cache at the beginning presumably to solve some other problem)